### PR TITLE
Passport integration

### PR DESCRIPTION
**Pull request by mKleinCreative**

### CODE REVIEW

File: source/server/api/v1/routes.js [hunk 1]
```diff
@@ -14,6 +15,21 @@ import {
...
+  [
+    'https://www.googleapis.com/auth/plus.login',
+    'https://www.googleapis.com/auth/plus.profile.emails.read'
+  ]
+}))
+
+router.get( '/google/auth/callback',
+  passport.authenticate( 'google', { failureRedirect: '/' } ),
+  ( request, response ) => {
+    response.status(200).json({ message: 'shit failed' })
```
Comment (nodatall) [inline source/server/api/v1/routes.js]: Status code 200 for failure? Also, be descriptive with the message so the future humans will not cry when they get errors.

File: source/dataServices/database/migrations/20170328100733_alter-user-for-oauth.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (fryymann715) [inline source/dataServices/database/migrations/20170328100733_alter-user-for-oauth.js]: removed

File: source/dataServices/database/migrations/20170328134726_add-display-name-to-user-table.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+exports.up = knex =>
+  knex.schema.table('users', table =>
+    table.varchar( 'displayName' ) )
+
+
+exports.down = knex =>
+  knex.schema.table('users', table =>
+    table.dropColumn( 'displayName' ) )
```
Comment (fryymann715) [inline source/dataServices/database/migrations/20170328134726_add-display-name-to-user-table.js]: There can be only one..

File: source/dataServices/database/queries/utilities.js [hunk 4]
```diff
@@ -1,5 +1,16 @@
 import knex from '../knex'
 
+const getUserByOAuthID = ( oauthID, callback ) =>
```
Comment (fryymann715) [inline source/dataServices/database/queries/utilities.js]: the first date went really well

File: source/dataServices/database/queries/utilities.js [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+  knex.table( 'users' ).where( 'oauthID', oauthID )
+    .then( user => {
+        if ( user == undefined ) {
```
Comment (fryymann715) [inline source/dataServices/database/queries/utilities.js]: bang user, check

File: source/dataServices/database/queries/utilities.js [hunk 6]
```diff
@@ -1,5 +1,16 @@
...
+          throw new Error( 'getUserByOAuthID: no record of user with oauthid ${oauthID} in database.' )
+        } else {
+          return user
+        }
+    })
+    .catch( error => callback( error ) )
```
Comment (fryymann715) [inline source/dataServices/database/queries/utilities.js]: error returned!

File: source/server/api/v1/routes.js [hunk 7]
```diff
@@ -14,6 +15,21 @@ import {
 
 const router = express()
 
+router.get(
+  '/auth/google',
+  passport.authenticate( 'google', { scope:
```
Comment (fryymann715) [inline source/server/api/v1/routes.js]: Moved as much as we could.

File: source/server/api/v1/routes.js [hunk 8]
```diff
@@ -14,6 +15,21 @@ import {
...
+  [
+    'https://www.googleapis.com/auth/plus.login',
+    'https://www.googleapis.com/auth/plus.profile.emails.read'
+  ]
+}))
+
+router.get( '/google/auth/callback',
+  passport.authenticate( 'google', { failureRedirect: '/' } ),
```
Comment (fryymann715) [inline source/server/api/v1/routes.js]: ✔️

File: source/server/api/v1/routes.js (hunk 1 above)
Comment (fryymann715) [inline source/server/api/v1/routes.js]: ✔️

**Pull request closed without merging**